### ISSUE TITLE
Could netcdf4 be pulled out to isolate GPL code?

### ISSUE THREAD

**Issue by danielballan** [intake/intake-xarray#41]
I think `netcdf4` is inflected with GPL code. I'm wondering if it could be made an optional dependency of `intake-xarray`, or moved into a separate `intake-netcdf4` package.

Although [conda-forge reports that netcdf4 has an MIT liencse](https://github.com/conda-forge/netcdf4-feedstock/blob/master/recipe/meta.yaml#L49), netcdf4 has a [hard dependency on `cftime`](https://github.com/Unidata/netcdf4-python/blob/master/setup.py#L7) which has a [GPL license](https://github.com/Unidata/cftime/blob/master/COPYING#L44).

h/t @dylanmcreynolds who noticed this problem.

**Comment by danielballan** [comment]
To be clear, I would be happy to do the work to make this happen if the maintainers of this package support the change.

**Comment by martindurant** [comment]
It would be ok by me to have this only as an optional dependency, but, of course, the docs had better be clear (and good error message on import failure).

**Comment by jsignell** [comment]
Yeah I am consistently surprised by the dependency on netcdf4. xarray doesn't depend on netcdf4, so we can probably look there for good error messaging.

**Comment by andersy005** [comment]
It's my understanding that the license inccompatibility issue in netcdf4-ptyhon/cftime got fixed a few days ago: https://github.com/Unidata/netcdf4-python/issues/1073.

**Comment by jsignell** [comment]
Seems like the conscensus here is that it would be fine to remove the xarray dep, but I'm not sure whether there is any motivation for doing that if the license issue is resolved.
